doc/api/vm.md

```diff
@@ -1124,10 +1142,10 @@ changes:
     [Support of dynamic `import()` in compilation APIs][].
 * Returns: {Object} contextified object.
```

Comment:
When created a vanilla context, the returned value is not behaving identically to general contextified objects. The returned vanilla context can retrieve the inner intrinsics directly, without via `vm.runInContext('globalThis', context).Object`.

```js
const vanilla = vm.createContext(vm.constants.VANILLA_CONTEXT);
const context = vm.createContext();

assert.strictEqual(typeof vanilla.Object, 'function');
assert.strictEqual(typeof context.Object, 'undefined');
```

I think this might be confusing to overload the function since it can be considered as a non-trivial difference.

Reply:
I wonder if documenting the differences would be enough in this case. The advantage of accepting it in the same API is that this can be reused as `runInNewContext()` arguments. If we invent specifically a new API for the creation, we might need to duplicate a new set of APIs (`runInNewVanillaContext`...) that specifically takes this variant, and that doubles the amount of documentation and tests, which doesn't seem to be worth it for this minor difference...

Reply:
Updated the docs a bit to stress that the built-ins can be directly accessed from the returned object.